src/plugin.tsx

```diff
@@ -30,54 +31,82 @@ interface KalturaLivePluginConfig {
     checkLiveWithKs: boolean;
 }
 
+export enum LiveBroadcastStates {
+    Unknown = "Unknown",
+    Live = "Live",
+    Offline = "Offline"
+}
+
 export class KalturaLivePlugin implements OnMediaUnload, OnRegisterUI, OnMediaLoad, OnPluginSetup {
     private _kalturaClient = new KalturaClient();
+    // represents if the entry type is a live entry
+    private _isLiveEntry = false;
+    // represents if the entry is in live mode or unknown
+    private _broadcastState: LiveBroadcastStates = LiveBroadcastStates.Unknown;
 
     constructor(
         private _contribServices: ContribServices,
         private _configs: ContribPluginConfigs<KalturaLivePluginConfig>,
         private _player: KalturaPlayerTypes.Player
-    ) {}
-
-    onPluginSetup(): void {
+    ) {
         const { playerConfig, pluginConfig } = this._configs;
         this._kalturaClient.setOptions({
             clientTag: "playkit-js-kaltura-live",
             endpointUrl: playerConfig.provider.env.serviceUrl
         });
-
         if (pluginConfig.checkLiveWithKs === true) {
             this._kalturaClient.setDefaultRequestOptions({
                 ks: playerConfig.provider.ks
             });
         }
+        // once we have the source we can tell if this is a live entry
         this._player.addEventListener(this._player.Event.SOURCE_SELECTED, this._isEntryLiveType);
-        this._isEntryLiveType();
     }
 
+    // implementations methods - todo - implement if necessary
+    onPluginSetup(): void {}
+
     onRegisterUI(uiManager: UIManager): void {}
 
     onMediaLoad(): void {}
 
     onMediaUnload(): void {}
 
-    private _isEntryLiveType = () => {
-        const { playerConfig } = this._configs;
+    public active() {
+        return this.isLiveEntry();
+    }
 
-        // TODO serhii please refactor
+    public isLiveEntry(): boolean {
+        return this._isLiveEntry;
+    }
 
-        // if (config.sources && config.sources.type === this._player.MediaType.LIVE) {
-        //     // this is Live
-        //     this._checkIsLive(use here the config object);
-        // }
+    public get player() {
+        return this._player;
+    }
+
+    // value of last 'isLive' API call
+    public get broadcastState(): LiveBroadcastStates {
+        return this._broadcastState;
+    }
+
+    private _isEntryLiveType = () => {
+        const { playerConfig } = this._configs;
+        if (playerConfig && playerConfig.sources.type === "Live") {
+            this._isLiveEntry = true;
+            this._checkIsLive();
```

Comment:
the names here are a bit confusing. usually method that starts with `is...` doesn't actually trigger any logic it only query existing state.